qa/1602

```diff
@@ -56,7 +54,7 @@ _save_config $PCP_SYSCONF_DIR/pmproxy
 $sudo rm -f $PCP_SYSCONF_DIR/pmproxy/*
 need_restore=true
 
-echo "== start two test Redis servers"
+echo "== start a test Redis server"
```

Comment:
@andreasgerstmayr I'm not following this - can we no longer specify multiple Redis servers this way?

Reply:
There's a couple of concerns with only supporting clusters now for multi-node setups:

- https://redis.io/topics/cluster-tutorial states: "Currently Redis Cluster does not support NATted environments and in general environments where IP addresses or TCP ports are remapped." ... not sure how big a deal that is in practice (that page does seem a little out of date maybe in places).

- we lose the ability to easily test and debug certain aspects (like key / command distribution, proxying - as in this test) - setting up an entire cluster is non-trivial relative to setting up just another Redis server.  This change to this test shows the problem there - its difficult to setup a cluster so the multi-server aspect was removed entirely.  :-(  This is an unfortunate backward step.

I guess it may be just the price we have to pay here, but ... it's not ideal.

Reply:
> There's a couple of concerns with only supporting clusters now for multi-node setups:
> 
> * https://redis.io/topics/cluster-tutorial states: "Currently Redis Cluster does not support NATted environments and in general environments where IP addresses or TCP ports are remapped." ... not sure how big a deal that is in practice (that page does seem a little out of date maybe in places).

Yep, Redis has now `cluster-announce-ip` and `cluster-announce-port` settings (see https://github.com/redis/redis/commit/5ac5e3ebd7da8f041ebd7c4d20020208a3066b2d), so I'd say this is resolved.

> * we lose the ability to easily test and debug certain aspects (like key / command distribution, proxying - as in this test) - setting up an entire cluster is non-trivial relative to setting up just another Redis server.  This change to this test shows the problem there - its difficult to setup a cluster so the multi-server aspect was removed entirely.  :-(  This is an unfortunate backward step.

Please take a look at qa/1603 - it creates a cluster, fills it with data from a PCP archive and also checks key distribution on each node (using `dbsize`). I removed the multi-node aspect from qa/1602 because multi-node cluster without cluster mode is not supported anymore, and we have qa/1603. But I can create a cluster in qa/1602 to test multi-node again if you wish.

Reply:
@andreasgerstmayr great!